kernel/src/main/java/org/kframework/kompile/Kompile.java

```diff
@@ -154,10 +154,24 @@ public CompiledDefinition run(File definitionFile, String mainModuleName, String
 
         if (kompileOptions.experimental.genBisonParser || kompileOptions.experimental.genGlrBisonParser) {
             new KRead(kem, files, InputModes.PROGRAM).createBisonParser(def.programParsingModuleFor(def.mainSyntaxModuleName(), kem).get(), def.programStartSymbol, files.resolveKompiled("parser_PGM"), kompileOptions.experimental.genGlrBisonParser);
-            for (Module mod : iterable(def.getParsedDefinition().entryModules())) {
-              if (mod.att().contains("parser")) {
-                new KRead(kem, files, InputModes.PROGRAM).createBisonParser(def.programParsingModuleFor(mod.name(), kem).get(), def.configurationVariableDefaultSorts.getOrDefault("$" + mod.att().get("parser"), Sorts.K()), files.resolveKompiled("parser_" + mod.att().get("parser")), kompileOptions.experimental.genGlrBisonParser);
-              }
+            for (Production prod : iterable(kompiledDefinition.mainModule().productions())) {
+                if (prod.att().contains("cell") && prod.att().contains("parser")) {
+                    String att = prod.att().get("parser");
+                    String[] parts = att.trim().split(";");
+                    for (String part : parts) {
+                        String[] subparts = part.trim().split(",");
```

Comment:
I believe we use this same strategy for getting subparts of many axioms. Can you look and see if it's reasonable to factor out a method directed at it?